audit/agent_log_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 Mia srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package audit
+
+import (
+	"context"
+	"testing"
+
+	rondlogrus "github.com/rond-authz/rond/logging/logrus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/sirupsen/logrus/hooks/test"
+)
+
+func TestLogAgent(t *testing.T) {
+	l, hook := test.NewNullLogger()
+	agent := NewLogAgent(rondlogrus.NewLogger(l))
+
+	agent.Trace(context.TODO(), Audit{
+		AggregationID: "the-aggregation-id",
+		Authorization: AuthzInfo{
+			Allowed:    true,
+			PolicyName: "some_policy",
+		},
+		Subject: SubjectInfo{
+			ID:     "some user",
+			Groups: []string{"g1", "g2"},
+		},
+		RequestBody: []byte("some body"),
+	})
+
+	entries := hook.AllEntries()
+	require.Len(t, entries, 1)
+	require.Equal(t, "audit trail", entries[0].Message)
```

Comment:
Why not assert also on the fields of the log?

Reply:
because while I was writing the test it felt pretty redundant and fragile

Reply:
It give a documentation of the output log, otherwise it is not clear

Reply:
but I'd have to serialize the log back to the Audit struct 🤷

Reply:
I've added the test, give it a look

Reply:
Perfect